src/main/java/com/example/petbuddybackend/entity/notification/ClientNotification.java

```diff
@@ -23,4 +24,8 @@ public class ClientNotification extends Notification {
     @JoinColumn(name = "clientEmail", nullable = false, updatable = false)
     private Client client;
 
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "triggeredBy", nullable = false, updatable = false)
+    private Caretaker triggeredBy;
```

Comment:
Ja bym dał nazwę żeby sugerowało że to opiekun ztriggerrował np. caretakerTrigger